src/main/java/bio/terra/service/resourcemanagement/google/GoogleBucketService.java

```diff
@@ -348,38 +370,47 @@ private Bucket newCloudBucket(
 
     GoogleProjectResource projectResource = bucketResource.getProjectResource();
     String googleProjectId = projectResource.getGoogleProjectId();
-    GcsProject gcsProject = gcsProjectFactory.get(googleProjectId);
+    GcsProject gcsProject = gcsProjectFactory.get(googleProjectId, true);
 
     // the project will have been created before this point, so no need to fetch it
     logger.info("Creating bucket '{}' in project '{}'", bucketName, googleProjectId);
     Storage storage = gcsProject.getStorage();
     Bucket createdBucket = storage.create(bucketInfo);
 
-    grantBucketReaderIam(createdBucket, storage, readerGroups);
+    grantBucketReaderIam(createdBucket, storage, readerGroups, dedicatedServiceAccount);
 
     return createdBucket;
   }
 
-  public void grantBucketReaderIam(Bucket bucket, Storage storage, List<String> readerGroups) {
-    if (readerGroups != null && !readerGroups.isEmpty()) {
-      String bucketName = bucket.getName();
+  public void grantBucketReaderIam(
+      Bucket bucket, Storage storage, List<String> readerGroups, String dedicatedServiceAccount) {
+    String bucketName = bucket.getName();
+
+    // Not sure why it needs to be version 3, but that's what the Google documentation says.
+    // https://cloud.google.com/storage/docs/access-control/using-iam-permissions#code-samples
+    Policy originalPolicy =
+        storage.getIamPolicy(bucketName, Storage.BucketSourceOption.requestedPolicyVersion(3));
+    List<Binding> bindings = new ArrayList<>(originalPolicy.getBindingsList());
 
-      // Not sure why it needs to be version 3, but that's what the Google documentation says.
-      // https://cloud.google.com/storage/docs/access-control/using-iam-permissions#code-samples
-      Policy originalPolicy =
-          storage.getIamPolicy(bucketName, Storage.BucketSourceOption.requestedPolicyVersion(3));
-      List<Binding> bindings = new ArrayList<>(originalPolicy.getBindingsList());
+    if (readerGroups != null) {
```

Comment:
We should never have to compare a collection against `null`, at least when dealing with our own APIs. Instead of `null`, the caller should pass an empty list.

Reply:
As  read the code, the `null` value doesn't come from JSON. It comes from this code in `newCloudBucket()`:
```
      if (getReaderGroups != null) {
        readerGroups = getReaderGroups.call();
      } else {
        readerGroups = null;
      }
``

Reply:
Regarding the code that `call()` invokes, I found six non-null cases and all were streaming calls that end in `toList()`.

Reply:
Regarding a general philosophy of bulletproofing code against invalid inputs, I think the best practice is to validate and normalize data close to the "outside" of our APIs, so data that arrives at the "inside" of our APIs is known to be safe.